widgetastic_manageiq.py

```diff
@@ -878,3 +875,57 @@ def last_page(self):
     @property
     def items_amount(self):
         return self.paginator.page_info()[1]
+
+
+class AlertEmail(Widget):
+
+    ROOT = "//div[@id='edit_email_div']"
+    ADD_BUTTON = ".//div[@title='Add']"
+    RECIPIENTS = "./div[@id='edit_to_email_div']//a"
+    EMAIL_LINK = ".//a[text()='{}']"
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.RECIPIENTS_INPUT = TextInput(self, "email")
+
+    def __locator__(self):
+        return self.ROOT
+
+    def fill(self, values):
+        if isinstance(values, basestring):
+            values = [values]
+        elif isinstance(values, list):
+            pass
+        else:
+            raise ValueError("Emails should be a string or a list")
+        if self.read() == set(values):
+            return False
+        else:
+            values_to_remove = self._values_to_remove(values)
+            values_to_add = self._values_to_add(values)
+            for value in values_to_remove:
+                self._remove_recipient(value)
+            for value in values_to_add:
+                self._add_recipient(value)
+            return True
+
+    @property
+    def button_add_recipient(self):
+        return self.browser.element(self.ADD_BUTTON, parent=self)
+
+    def _values_to_remove(self, values):
+        return list((set(values) ^ self.read()) - set(values))
```

Comment:
@quarckster, if I understand this right, you want to remove recipents which exist in read() but don't exist in values.
you could just do self.read() - set(values). I possible miss something.
Could you please explain what is xor for there ?